x/foundation/authz.go

```diff
@@ -0,0 +1,87 @@
+package foundation
+
+import (
+	"github.com/gogo/protobuf/proto"
+
+	sdk "github.com/line/lbm-sdk/types"
+	sdkerrors "github.com/line/lbm-sdk/types/errors"
+	stakingtypes "github.com/line/lbm-sdk/x/staking/types"
+)
+
+// Authorization represents the interface of various Authorization types implemented
+// by other modules.
+// Caution: It's not for x/authz exec.
+type Authorization interface {
+	proto.Message
+
+	// MsgTypeURL returns the fully-qualified Msg service method URL (as described in ADR 031),
+	// which will process and accept or reject a request.
+	MsgTypeURL() string
+
+	// Accept determines whether this grant permits the provided sdk.Msg to be performed,
+	// and if so provides an updated authorization instance.
+	Accept(ctx sdk.Context, msg sdk.Msg) (AcceptResponse, error)
+
+	// ValidateBasic does a simple validation check that
+	// doesn't require access to any other information.
+	ValidateBasic() error
+}
+
+// AcceptResponse instruments the controller of an authz message if the request is accepted
+// and if it should be updated or deleted.
+type AcceptResponse struct {
+	// If Accept=true, the controller can accept and authorization and handle the update.
+	Accept bool
+	// If Delete=true, the controller must delete the authorization object and release
+	// storage resources.
+	Delete bool
+	// Controller, who is calling Authorization.Accept must check if `Updated != nil`. If yes,
+	// it must use the updated version and handle the update on the storage level.
+	Updated Authorization
+}
```

Comment:
1. It only shares the interface, but it has nothing to do with x/authz. To avoid the confusion, I decided to copy it.
2. It makes x/foundation and other module to depend on x/authz.

I have copied it because of our discussion. You can, of course, ask me to revert it.

Reply:
I understand it. Thank you. If then, there is no need to follow the change of authz's `Authorization` interface? If we need to apply the change of authz in the future, I think it's a better to use authz's interface not copy.

Reply:
It's different from that of x/authz, so we don't need to use it anymore.